src/components/campaignguide/ScenarioView.tsx

```diff
@@ -173,22 +173,40 @@ function ScenarioView({ componentId, campaignId, showLinkedScenario, processedSc
     const campaignCode = processedScenario.scenarioGuide.campaignGuide.campaignCycleCode();
     const campaignName = processedScenario.scenarioGuide.campaignGuide.campaignName();
     const scenarioName = processedScenario.scenarioGuide.scenarioName();
+
     const queue: NarrationTrack[] = [];
+    const scenarioSteps: ScenarioStep[] = [];
     for (const scenarioStep of processedScenario.steps) {
+      if (scenarioStep.step.type === 'effects') {
+        for (const effectsWithInput of scenarioStep.step.effectsWithInput) {
+          for (const effect of effectsWithInput.effects) {
+            if ('steps' in effect) {
+              scenarioSteps.push(...processedScenario.scenarioGuide.expandSteps(
+                effect.steps,
+                scenarioState,
+                scenarioStep.campaignLog
+              ));
+            }
+          }
+        }
+      } else {
+        scenarioSteps.push(scenarioStep);
+      }
+    }
+    for (const scenarioStep of scenarioSteps) {
       if (scenarioStep.step.type === 'resolution') {
         const narration = processedScenario.scenarioGuide.resolution(
           scenarioStep.step.resolution
         )?.narration;
-        if (!narration) {
-          continue;
-        }
+        if (!narration) continue;
+
         queue.push({
           ...narration,
           campaignCode,
           campaignName,
           scenarioName,
         });
-      } else if (scenarioStep.step.type === 'story' || scenarioStep.step.type === 'branch') {
+      } else if (['story', 'branch', 'input'].includes(scenarioStep.step.type ?? '')) {
```

Comment:
It's small, but I might make this a constant `Set` rather than constructing and doing list search everytime

Reply:
Whoops, used to Dart / Flutter where I could add `const` to it and have it do it for me :P